docs/src/literate-tutorials/reactive_surface.jl

```diff
@@ -0,0 +1,259 @@
+function setup_initial_conditions!(u₀::Vector, cellvalues::CellValues, dh::DofHandler)
+    u₀ .= ones(ndofs(dh))
+    u₀[2:2:end] .= 0.0
+
+    n_basefuncs = getnbasefunctions(cellvalues)
+
+    for cell in CellIterator(dh)
+        reinit!(cellvalues, cell)
+
+        coords = getcoordinates(cell)
+        dofs = celldofs(cell)
+        uₑ = @view u₀[dofs]
+        rv₀ₑ = reshape(uₑ, (2, n_basefuncs))
+
+        for i in 1:n_basefuncs
+            if coords[i][3] > 0.9
+                rv₀ₑ[1, i] = 0.5
+                rv₀ₑ[2, i] = 0.25
+            end
+        end
+    end
+
+    u₀ .+= 0.01*rand(ndofs(dh))
+end
+
+# ### Simulation routines
+# Now we define a function to setup and solve the problem with given feed and conversion rates
+# $F$ and $k$, as well as the time step length and for how long we want to solve the model.
+function gray_scott_sphere(F, k, Δt, T, refinements)
+    ## We start by setting up grid, dof handler and the matrices for the heat problem.
+    gmsh.initialize()
+
+    ## Add a unit sphere
+    gmsh.model.occ.addSphere(0.0,0.0,0.0,1.0)
+    gmsh.model.occ.synchronize()
+
+    gmsh.model.mesh.renumberNodes()
+    gmsh.model.mesh.renumberElements()
+
+    ## Generate surface elements and refine several times
+    gmsh.model.mesh.generate(2)
+    for _ in 1:refinements
+        gmsh.model.mesh.refine()
+    end
+
+    ## Create a grid out of it
+    nodes = tonodes()
+    elements, _ = toelements(2)
+    gmsh.finalize()
+    grid = Grid(elements, nodes);
+
+    ip = Lagrange{RefTriangle, 1}()
+    ip_geo = Lagrange{RefTriangle, 1}()
+    qr = QuadratureRule{RefTriangle}(2)
+    cellvalues = CellValues(qr, ip, ip_geo^3);
+
+    dh = DofHandler(grid);
+    add!(dh, :reactants, ip^2);
+    close!(dh);
+
+    M = create_sparsity_pattern(dh; coupling=[true false;false true])
+    D = create_sparsity_pattern(dh; coupling=[true false;false true])
+    assemble_matrices!(M, D, cellvalues, dh);
+
+    ## Since the heat problem is linear and has no time dependent parameters, we precompute the
+    ## decomposition of the system matrix to speed up the linear system solver.
+    A = M + Δt .* D
+    Alu = cholesky(A)
+
+    ## Now we setup buffers for the time dependent solution and fill the initial condition.
+    uₜ   = zeros(ndofs(dh))
+    uₜ₋₁ = ones(ndofs(dh))
+    setup_initial_conditions!(uₜ₋₁, cellvalues, dh)
+
+    ## And prepare output for visualization.
+    pvd = VTKFileCollection("reactive-surface.pvd", dh);
+    addstep!(pvd, 0.0) do io
+        write_solution(io, dh, uₜ₋₁)
+    end
+
+    ## This is now the main solve loop.
+    for (iₜ, t) ∈ enumerate(Δt:Δt:T)
+        ## First we solve the heat problem
+        uₜ .= Alu \ (M * uₜ₋₁)
+
+        ## Then we solve the point-wise reaction problem with the solution of
+        ## the heat problem as initial guess.
+        rvₜ = reshape(uₜ, (2, length(grid.nodes)))
+        for i ∈ 1:length(grid.nodes)
```

Comment:
This somehow feels a bit iffy... but maybe we don't have a better way to do this now. Would
```suggestion
        rvₜ = reshape(uₜ, (2, ndofs(dh) ÷ 2))
        for i ∈ axes(rvₜ, 2)
```
be equivalent?

Reply:
I expanded the explanation on this, is it more clear now?